aten/src/ATen/templates/TensorBody.h

```diff
@@ -147,6 +147,10 @@ class TORCH_API Tensor {
     return impl_;
   }
 
+  c10::intrusive_ptr<TensorImpl, UndefinedTensorImpl> unsafeReleaseIntrusivePtr()  {
```

Comment:
have you considered instead calling it
```
c10::intrusive_ptr<TensorImpl, UndefinedTensorImpl> getIntrusivePtr() && {
  return std::move(impl_);
}
```

Then the call site can be `std::move(data).getIntrusivePtr()`. Not sure it's better but I've seen this pattern in other libraries (e.g. https://github.com/apache/arrow/blob/master/cpp/src/arrow/result.h#L84 )

Reply:
That would be inconsistent with `unsafeReleaseTensorImpl()`, above..